packages/client/components/BottomControlBarReadyButton.tsx

```diff
@@ -0,0 +1,169 @@
+  const meeting = useFragment(
+    graphql`
+      fragment BottomControlBarReadyButton_meeting on NewMeeting {
+        localPhase {
+          phaseType
+        }
+        ... on RetrospectiveMeeting {
+          endedAt
+          facilitatorUserId
+          localPhase {
+            ... on ReflectPhase {
+              reflectPrompts {
+                id
+                editorIds
+              }
+            }
+          }
+          reflectionGroups {
+            id
+            reflections {
+              id
+              isEditing
+              isDropping
+              isViewerDragging
+            }
+          }
+          viewerMeetingMember {
+            votesRemaining
+          }
+          votesRemaining
+        }
+      }
+    `,
+    meetingRef
+  )
+  const {
+    endedAt,
+    facilitatorUserId,
+    localPhase,
+    reflectionGroups,
+    viewerMeetingMember,
+    votesRemaining
+  } = meeting
+  const [delaySeconds, setDelaySeconds] = useState<number>()
+  const atmosphere = useAtmosphere()
+  const {viewerId} = atmosphere
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'reflect') return
+
+    const hasNoReflection = reflectionGroups?.every((group) => group.reflections?.length === 0)
+    if (hasNoReflection) return
+
+    const reflectPrompts = localPhase.reflectPrompts!
+    const beforeDemoStart =
+      isDemoRoute() && reflectPrompts.every(({editorIds}) => !Array.isArray(editorIds))
+    if (beforeDemoStart) return
+
+    const isNotEditing = reflectPrompts.every(({editorIds}) => {
+      return editorIds === undefined || (Array.isArray(editorIds) && editorIds.length === 0)
+    })
+
+    setDelaySeconds(isNotEditing ? 30 : undefined)
+  }, [localPhase.phaseType === 'reflect', reflectionGroups, localPhase])
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'group') return
+
+    const isNotDragging = reflectionGroups?.every((group) =>
+      group.reflections?.every(
+        (reflection) => !reflection.isViewerDragging && !reflection.isDropping
+      )
+    )
+
+    setDelaySeconds(isNotDragging ? 30 : undefined)
+  }, [localPhase.phaseType === 'group', reflectionGroups, localPhase])
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'vote') return
+
+    const teamVotesRemaining = votesRemaining || 0
+    const myVotesRemaining = viewerMeetingMember?.votesRemaining || 0
+    const isNotVoting = teamVotesRemaining === 0 || myVotesRemaining === 0
+
+    setDelaySeconds(isNotVoting ? 30 : undefined)
+  }, [localPhase.phaseType === 'vote', votesRemaining, viewerMeetingMember])
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'discuss') return
+
+    setDelaySeconds(5 * 60)
+  }, [localPhase.phaseType === 'discuss'])
+
+  useEffect(() => {
+    // if the ready button is "full" before these conditions are met,
+    // the animation should start after 5s
+    if (progress !== 1) return
+
+    setDelaySeconds(5)
+  }, [progress, localPhase.phaseType])
```

Comment:
-1 why did you add `localPhase.phaseType` here?
+1 you could optimize the dependency array by `[progress === 1]`